pkg/controller/controller_test.go

```diff
@@ -230,6 +232,24 @@ func TestRolloutController_Reconcile(t *testing.T) {
 			// the first batch of zone-a pods is deleted
 			expectedDeletedPods: []string{"ingester-zone-a-0", "ingester-zone-a-1"},
 		},
+		"zones a & b need full updates, the zpdb does not deny anything in zone-a - partition mode": {
+			statefulSets: []runtime.Object{
+				mockStatefulSet("ingester-zone-a", withPrevRevision()),
+				mockStatefulSet("ingester-zone-b", withPrevRevision()),
+			},
+			pods: []runtime.Object{
+				mockStatefulSetPod("ingester-zone-a-0", testPrevRevisionHash),
+				mockStatefulSetPod("ingester-zone-a-1", testPrevRevisionHash),
+				mockStatefulSetPod("ingester-zone-a-2", testPrevRevisionHash),
+				mockStatefulSetPod("ingester-zone-b-0", testPrevRevisionHash),
+				mockStatefulSetPod("ingester-zone-b-1", testPrevRevisionHash),
+				mockStatefulSetPod("ingester-zone-b-2", testPrevRevisionHash),
+			},
+			// the first batch of zone-a pods is deleted
+			expectedDeletedPods: []string{"ingester-zone-a-0", "ingester-zone-a-1"},
+			// setting this will trigger the maxUnavailability override if 0 to be 1
```

Comment:
Doesn't ZPDB partition mode cause the maxUnavailability override to be _unconditionally_ 1?
```suggestion
			// setting this will trigger the maxUnavailability override to be 1
```